### PR TITLE
Adding the Collections component of the Art Gallery

### PR DESCRIPTION
**Pull request by adtsim**
**Description**
This pull request introduces three new React components to enhance the functionality and user experience of our art gallery application. The new components are CollectionDetails, Collections, and CompareArtworks. Below is a detailed overview of each component and their functionalities.

**Components Added**

**1. CollectionDetails.jsx**
  **Purpose**: To display detailed information about a specific collection.
  **Features:** 
-   Fetches collection details using the collectionId parameter from the URL.
-   Displays the collection's title, image, and description.
-   Handles loading states and errors gracefully.

**2. Collections.jsx**
  **Purpose:** To display a list of all available collections.
  **Features:**
-   Fetches and lists all collections.
-   Provides links to detailed views of each collection.
-   Handles loading states and errors gracefully.

**3. CompareArtworks.jsx**
**Purpose:** To compare artworks across different centuries and display the data in a bar chart.
**Features:**
- Fetches and processes artwork data by type and period.
- Formats the data for use in a bar chart using react-chartjs-2.
- Displays the comparison of artworks across centuries in a visually appealing bar chart.
- Handles loading states and errors gracefully.

### CODE REVIEW

File: src/services/apiService.js [hunk 1]
```diff
@@ -171,13 +171,54 @@ export const getPieceById = async (id) => {
...
 };
 
+export const fetchCollections = async () => {
+  try {
+    // Request a single page of 100 painting items from the Rijksmuseum API
+    const response = await axios.get(
+      `https://www.rijksmuseum.nl/api/en/collection`,
+      {
+        params: {
+          key: "A3B8HWv5",
```
Comment (meghanasharathd) [inline src/services/apiService.js]: API key is hardcoded

Comment (SiddharthSubash) [inline src/services/apiService.js]: That is not a good practise to hardcode Api keys. Kindly change that

File: src/components/Collections.jsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    <div
+      style={{
+        display: "flex",
+        justifyContent: "space-between",
+        padding: "20px",
+        alignItems: "flex-start", // Align items at the top
+      }}
+    >
+      <div style={{ width: "40%", overflowY: "auto", maxHeight: "400px" }}>
+        <h1>Artists</h1>
```
Comment (SiddharthSubash) [inline src/components/Collections.jsx]: The artist title as well as bargraph labels and x axis are too greyed out.

Use a brighter color to represent them

File: src/components/Collections.jsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+      setMakersData(makersCount);
+    };
+    getCollections();
+  }, []);
+
+  const data = {
+    labels: Object.keys(makersData),
+    datasets: [
+      {
+        label: "Number of Artworks",
```
Comment (SiddharthSubash) [inline src/components/Collections.jsx:56]: Same with here the label is not at all visible. Use a color that is contrasting with the background color.

File: src/components/Collections.jsx [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        </ul>
+      </div>
+      <div style={{ width: "55%" }}>
+        <Bar data={data} options={{ scales: { y: { beginAtZero: true } } }} />
+      </div>
+    </div>
+  );
+}
+
+export default Collections;
```
Comment (SiddharthSubash) [inline src/components/Collections.jsx:154]: There is list of artists below the title, which I am unable to see until I hover over it. Change the design in a way that I should be able to see the list of artists without hovering it.

Changing the color of the text might resolve this.

File: src/components/Collections.jsx [hunk 5]
```diff
@@ -0,0 +1,110 @@
+import { useEffect, useState } from "react";
```
Comment (SiddharthSubash) [inline src/components/Collections.jsx:2]: Adhered well to the proper coding style guidelines like Airbnb, bootstrap and cleaned code guidance.
The structure of the code is in compliant with coding semantic properly.
The code and view is properly getting resized based on the window size.

File: src/components/Collections.jsx [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+      <div style={{ width: "40%", overflowY: "auto", maxHeight: "400px" }}>
+        <h1>Artists</h1>
+        <ul style={{ listStyleType: "none", padding: 0 }}>
+          {Object.keys(makersData).map((maker, index) => (
+            <li key={index} style={{ marginBottom: "10px" }}>
+              <Link
+                to={`/collection-details/${encodeURIComponent(maker)}`}
+                style={linkStyle}
+                onMouseEnter={(e) =>
+                  (e.currentTarget.style.backgroundColor = "#f0f0f0")
```
Comment (SiddharthSubash) [inline src/components/Collections.jsx]: It is good that you have introduced hover whenever mouse hovers over the list of artists.

File: src/components/Collections.jsx [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+        alignItems: "flex-start", // Align items at the top
+      }}
+    >
+      <div style={{ width: "40%", overflowY: "auto", maxHeight: "400px" }}>
+        <h1>Artists</h1>
+        <ul style={{ listStyleType: "none", padding: 0 }}>
+          {Object.keys(makersData).map((maker, index) => (
+            <li key={index} style={{ marginBottom: "10px" }}>
+              <Link
+                to={`/collection-details/${encodeURIComponent(maker)}`}
```
Comment (meghanasharathd) [inline src/components/Collections.jsx:130]: Add aria-label attributes to the Link components for better accessibility.

File: src/components/CollectionDetails.jsx [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    };
+    getArtworks();
+  }, [maker]);
+
+  return (
+    <div>
+      <h1>Artworks by {decodeURIComponent(maker)}</h1>
+      {artworks.map((artwork) => (
+        <div key={artwork.id}>
+          <h2>{artwork.title}</h2>
```
Comment (SiddharthSubash) [inline src/components/CollectionDetails.jsx]: Here I am able to see the images but not the text that is describing about the images.

File: src/components/CompareArtworks.jsx [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+          <Bar dataKey="19th Century" fill="#8884d8" />
+          <Bar dataKey="20th Century" fill="#82ca9d" />
+          <Bar dataKey="21st Century" fill="#ffc658" />
+        </BarChart>
+      </ResponsiveContainer>
+    </div>
+  );
+}
+
+export default ArtworkComparison;
```
Comment (SiddharthSubash) [inline src/components/CompareArtworks.jsx:103]: Nicely done with comparing the artwork. Good design and followed the coding guidelines well.

File: src/components/CompareArtworks.jsx [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+} from "recharts";
+
+function ArtworkComparison() {
+  const [data, setData] = useState([]);
+  const [isLoading, setIsLoading] = useState(true); // Loading state
+
+  useEffect(() => {
+    const materials = ["chalk", "ink", "pencil"];
+    const centuries = { 18: 1700, 19: 1800, 20: 1900, 21: 2000 };
+    const loadData = async () => {
```
Comment (meghanasharathd) [inline src/components/CompareArtworks.jsx:20]: The use of async/await ensures clarity and readability in handling asynchronous operations.

File: src/components/CollectionDetails.jsx (hunk 8 above)
Comment (adtsim) [reply src/components/CollectionDetails.jsx]: Resolved in my latest commit, thanks!

File: src/components/Collections.jsx (hunk 5 above)
Comment (adtsim) [reply src/components/Collections.jsx:2]: Thank you!

File: src/components/Collections.jsx (hunk 3 above)
Comment (adtsim) [reply src/components/Collections.jsx:56]: Addressed, thanks!

File: src/components/Collections.jsx (hunk 2 above)
Comment (adtsim) [reply src/components/Collections.jsx]: Addressed, thank you!

File: src/components/Collections.jsx (hunk 6 above)
Comment (adtsim) [reply src/components/Collections.jsx]: Thank you!

File: src/components/Collections.jsx (hunk 4 above)
Comment (adtsim) [reply src/components/Collections.jsx:154]: Addressed in my latest commit, thanks!

File: src/components/CompareArtworks.jsx (hunk 9 above)
Comment (adtsim) [reply src/components/CompareArtworks.jsx:103]: Thanks!

File: src/services/apiService.js (hunk 1 above)
Comment (adtsim) [reply src/services/apiService.js]: Corrected this part of code in my latest commit, thanks for pointing it out!

Comment (adtsim) [reply src/services/apiService.js]: Addressed it as part of my latest changes, thanks!

File: src/components/Collections.jsx (hunk 7 above)
Comment (adtsim) [reply src/components/Collections.jsx:130]: Addressed in my latest commit, thank you!

File: src/components/CompareArtworks.jsx (hunk 10 above)
Comment (adtsim) [reply src/components/CompareArtworks.jsx:20]: Thanks!

**Review by SiddharthSubash — APPROVED** [review]

**Review by meghanasharathd — APPROVED** [review]

**Pull request merged**